### ISSUE TITLE
[Accessibility] Screen Reader Interactions

### ISSUE THREAD

**Issue by eileencho** [jupyter/notebook#5106]
Description: There are some areas in the notebook that are difficult to use with a screen reader. Sometimes there is information that is not exposed to the user, or the information is incorrect, or there exist items that the user should be able to access, but cannot when using a screen reader. 

Examples [with pictures](https://docs.google.com/document/d/1q_7LiqLX2WGJZ0UHNWDFafw75McZgLp9v4Ifdzn2FYE/edit#heading=h.8mvgk0mlzv3z):

- Menu bar items are announced as being out of 10 when there are only 8 items
- Submenu items such as File > New Notebook > Python 3 can never be accessed via screen reader
- The toggle buttons "Name", "Last Modified", "File Size" on the dashboard never announce their state of which direction they are sorting by

Reproduce:
1. Open Notebook dashboard to see the file directory
2. Download and start screen reader (such as [Voiceover](https://www.apple.com/voiceover/info/guide/_1124.html) which is already available on Macs, [NVDA](https://www.nvaccess.org/download/) on Windows, [Orca](https://wiki.gnome.org/Projects/Orca) for Linux)
3. Using the screen reader, navigate to the "Name", "Last Modified", "File Size"  toggle buttons and note that when using them, state is never announced. 
4. Separately, open a notebook, and navigate to the menu bar using the screen reader. Note the information announced when accessing each item, as well as how there are submenu items that cannot be reached.

[WCAG 2.0 Guidelines Section 4.1.2](https://www.w3.org/TR/UNDERSTANDING-WCAG20/ensure-compat-rsv.html)
[NYU Full Accessibility Audit](https://docs.google.com/spreadsheets/d/1iCiEExo4VDMOwAYtBTCdh2chyx394-1r6YwzmRILF9E/edit?usp=sharing) done with @VickySteeves

**Comment by balajialg** [comment]
@jtpio I tried testing the alpha version of Jupyter Notebook 7.0 using a [Screen reader tool](https://chrome.google.com/webstore/detail/screen-reader/kgejglhpjiefppelpmljglcjbhoiplfn) available as a Chrome plugin. I tested this functionality by launching the binder link of the alpha version of Jupyter Notebook 7.0 using a Chrome browser on a Windows machine. 

I found the above issue of screen readers not being able to convey the sub-menu information still persists with the alpha version of Notebook 7.0. The screen reader shows blank responses when I hover over the icons present as a sub-menu. It reads the menu icon with text correctly  (like the interface button) but if there is a menu without text then the screen reader throws a blank response.
![image](https://user-images.githubusercontent.com/2306166/197597583-aad26948-c213-4bcb-a460-6bf6d0a4a180.png)

The menu icons appearing in the cells executing code also are not decipherable for the screen readers. 
![image](https://user-images.githubusercontent.com/2306166/197597717-c432a957-1114-4ea9-90d8-abe9eaf5cfe5.png)

Reproduce:
- Launch the alpha version of Jupyter notebook 7.0 available at this [link](https://discourse.jupyter.org/t/notebook-7-pre-releases-are-available/16063)
- Install Screen reader plugin in Chrome browser
- Launch a new notebook
- Navigate to sub-menu icons and try hovering over them 
- Create a new cell in the notebook
- Navigate to the menu options at the right corner of the cell and hover over the icons

I guess this should get addressed and prioritized for the upcoming Notebook 7.0 release.  Let me know if you need more information. Thanks!
